src/components/Profile/EditProfile.js

```diff
@@ -15,233 +15,258 @@ import ProfileForm from './ProfileForm'
+  (
+    {
+      coverPic,
+      coverPicRemovalEnabled,
+      onBack,
+      onCoverPicChange,
+      onCoverPicRemoval,
+      profile,
+    },
+    coverPicInputRef
+  ) => {
+    const [selectedTab, setSelectedTab] = useState(0)
+    const [profilePic, onProfilePicChange, onProfilePicRemoval] = usePicture(
+      true
+    )
 
-    const formData = new FormData()
-    formData.append('path', newPhotoFile)
+    const theme = useTheme()
+    const { name: layout } = useLayout()
+    const { account, image, name } = profile || {}
+    const oneColumn = layout === 'small' || layout === 'medium'
 
-    setProfilePic({ buffer: formData, updated: true, removed: false })
-  }, [])
+    const imageInput = useRef(null)
 
-  const handleImageRemoval = useCallback(() => {
-    setProfilePic({ buffer: null, updated: false, removed: true })
-  }, [])
+    const [Content, props] = useMemo(() => {
+      const TabContent = CONTENT[selectedTab]
 
-  const handlePicFetch = useCallback(async buffer => {
-    const res = await fetch(IPFS_ENDPOINT.upload, {
-      method: 'post',
-      'Content-Type': 'multipart/form-data',
-      body: buffer,
-    })
+      const props = {}
+      if (selectedTab === 0) {
+        props.coverPic = coverPic
+        props.profile = profile
+        props.profilePic = profilePic
+      }
 
-    return res.json()
-  }, [])
+      return [TabContent, props]
+    }, [coverPic, profile, profilePic, selectedTab])
 
-  const [Content, props] = useMemo(() => {
-    const TabContent = CONTENT[selectedTab]
-
-    const props = {}
-    if (selectedTab === 0) {
-      props.fetchProfilePic = handlePicFetch
-      props.profile = profile
-      props.profilePic = profilePic
-    }
-
-    return [TabContent, props]
-  }, [handlePicFetch, profile, profilePic, selectedTab])
-
-  return (
-    <div>
-      <div
-        css={`
-          text-align: right;
-          margin-bottom: ${2 * GU}px;
-        `}
-      >
-        <Button label="Change background" />
-      </div>
-      <Split
-        primary={<Content onBack={onBack} {...props} />}
-        secondary={
+    return (
+      <div>
+        <div
+          css={`
+            display: flex;
+            justify-content: flex-end;
+            margin-bottom: ${2 * GU}px;
+          `}
+        >
+          {selectedTab === 0 && coverPicRemovalEnabled && (
+            <Button
+              label="Remove cover"
+              onClick={onCoverPicRemoval}
+              icon={<IconCross />}
+              display="icon"
+              css={`
+                border-radius: 50%;
+                margin-right: ${2 * GU}px;
+              `}
+            />
+          )}
           <div
             css={`
               position: relative;
-              text-align: center;
             `}
           >
+            <label htmlFor="coverPic">
+              <input
+                id="coverPic"
+                type="file"
+                accept="image/*"
```

Comment:
this accepts any kind of image extension? is it compatible with 3box?

Reply:
Yes! This is actually copied form the 3box dapp :D